tests/datadriven/test_reads.py

```diff
@@ -5,18 +5,58 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import collections
+import glob
+import os
+
 import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
 import ga4gh.datamodel.reads as reads
 import tests.datadriven as datadriven
 
+import pysam
+
 
-def testVariantSets():
+def testReads():
     testDataDir = "tests/data/reads"
     for test in datadriven.makeTests(testDataDir, ReadGroupSetTest):
         yield test
 
 
 class ReadGroupSetTest(datadriven.DataDrivenTest):
+    """
+    Data driven test for read group sets
+    """
+    class ReadGroupInfo(object):
+        """
+        Container class for information about a read group
+        """
+        def __init__(self, readGroupSetId, samFileName):
+            filename = os.path.split(samFileName)[1]
+            localId = os.path.splitext(filename)[0]
+            readGroupId = "{}:{}".format(readGroupSetId, localId)
+            self.id = readGroupId
+            self.samFile = pysam.AlignmentFile(samFileName)
+            self.reads = []
+            self.refNames = collections.defaultdict(list)
+            self.refIds = collections.defaultdict(list)
+            for read in self.samFile:
+                refName = self.samFile.getrname(read.reference_id)
+                self.refNames[refName].append(read)
+                refId = read.reference_id
+                self.refIds[refId].append(read)
+                self.reads.append(read)
+
+    def __init__(self, readGroupSetId, baseDir):
+        super(ReadGroupSetTest, self).__init__(readGroupSetId, baseDir)
+        self._readGroupInfos = {}
+        for samFileName in glob.glob(
+                os.path.join(self._dataDir, "*.bam")):
+            self._readSam(readGroupSetId, samFileName)
```

Comment:
why the private method call here? Wouldn't this single statement be equivalent and simpler?
```
self._readGroupInfos[samFileName] = self.ReadGroupInfo(
    readGroupSetId, samFileName)
```